pkg/prowgen/prowgen_tests.go

```diff
@@ -48,27 +52,29 @@ func DiscoverTests(r Repository, openShift OpenShift, sourceImageName string, sk
 				testTimeout = &prowapi.Duration{Duration: 4 * time.Hour}
 			}
 
-			clusterClaim := &cioperatorapi.ClusterClaim{
-				Product:      cioperatorapi.ReleaseProductOCP,
-				Version:      openShift.Version,
-				Architecture: cioperatorapi.ReleaseArchitectureAMD64,
-				Cloud:        cioperatorapi.CloudAWS,
-				Owner:        "openshift-ci",
-				Timeout:      &prowapi.Duration{Duration: time.Hour},
-			}
-			var clusterProfile cioperatorapi.ClusterProfile
-			workflow := pointer.String("generic-claim")
-			var env cioperatorapi.TestEnvironment
-			if openShift.CandidateRelease {
-				clusterClaim = nil
-				if strings.Contains(r.RepositoryDirectory(), "serverless-operator") {
-					// Use the shared profile for s-o.
-					clusterProfile = "aws"
-				} else {
-					clusterProfile = "aws-serverless"
-					env = map[string]string{
-						"BASE_DOMAIN": devclusterBaseDomain,
-					}
+			var (
+				clusterClaim   *cioperatorapi.ClusterClaim
+				clusterProfile cioperatorapi.ClusterProfile
+				workflow       *string
+				env            cioperatorapi.TestEnvironment
+			)
+
+			useClusterPool := openShift.Version == clusterPoolVersion
+			// Make sure to use the existing cluster pool if available for the given OpenShift version.
+			if useClusterPool {
+				clusterClaim = &cioperatorapi.ClusterClaim{
```

Comment:
urgh, that configuration is hard to parse (cause of prow I assume). Let me see if I get this right:
- if we want to use our pool, we set `clusterClaim` otherwise it is `nil`
- if we do not want to use our pool, we set `clusterProfile` to `aws-serverless` otherwise it is nil.
- we have to set workflow in both cases

I think same question as above, why do we refer to `aws-serverless` here when we are not using our pool. What does that define?

Reply:
You're correct.
* For cluster pools, we need to use `clusterClaim` and the workflow needs to be `generic-claim`
* For clusters starting from scratch, we need to use `clusterProfile` (this is mutually exclusive with `clusterClaim`) and the workflow needs to be `ipi-aws`
Regarding `aws-serverless`, this is answered in the other comment.

Reply:
If starting a cluster from scratch, we also need to reference a "post" step `ipi-deprovision-deprovision`